File: src/javaDemo/SwitchDemo.java
package javaDemo;

public class SwitchDemo {
    public static void main(String[] args) {
        Demo.switchTest(10);
        Demo.ifElseTest(10);
    }
}

class Demo {
    static void switchTest(int param) {
        String grade;
        switch (param) {
            case 10:
            case 9:
                grade = "Excellent";
                break;
            case 8:
            case 7:
            case 6:
                grade = "Good";
                break;
            case 5:
            case 4:
                grade = "Ok";
                break;
            case 3:
            case 2:
            case 1:
                grade = "Fail";
                break;
            default:
                grade = "Fail";
        }
        System.out.println("Grade is:" + grade);
    }

    static void ifElseTest(int count) {
        int result = count > 0 ? 10 : 0;
        System.out.println("Result is:" + result);
    }
}

File: src/kotlinDemo/SwitchDemo.kt
package kotlinDemo

fun main() {
    SwitchDemo.switchTest(10)
    SwitchDemo.whenTest()
    SwitchDemo.ifElseTest(1)
}

object SwitchDemo {
    fun switchTest(param: Int) {
        val grade = when (param) {
            9, 10 -> "Excellent"
            in 6..8 -> "Good"
            4, 5 -> "Ok"
            in 1..3 -> "Fail"
            else -> "Fail"
        }
        println("Grade is: $grade")
    }

    fun whenTest() {
        val x = 6
        val s = "hello"
        val res = when {
            x in 1..10 -> "cheap"
            s.contains("hello") -> "it's a welcome!"
            else -> ""
        }
        println("Result is: $res")
    }

    fun ifElseTest(count: Int) {
        val result = if (count > 0) 10 else 0
        println("Result is: $result")
    }
}